src/services/log.ts

```diff
@@ -432,11 +457,70 @@ export function makeCopyBtn(v: any, copyBag: Array<any>): string {
   return `<span class="icon copy-btn tooltip is-tooltip-right" data-tooltip="Copy" data-copy="${copyBag.length - 1}"><i class="mdi mdi-content-copy"></i></span>`
 }
 
-export const transformers: { [key: string]: (any) => Transform } = {
+interface JavaExceptionMatch {
+  // If a fqcn has this prefix, it will turn into a link.
+  fqcnMatch: string
+  // The link generated for a matched fqcn
+  href: string
+}
+
+interface JavaExceptionConfig {
+  matches: JavaExceptionMatch[]
+}
+
+function fqcnMatches(fqcn: string, config: JavaExceptionConfig): JavaExceptionMatch | undefined {
+  return config.matches.find(m => fqcn.match(new RegExp(m.fqcnMatch)))
+}
+
+function javaException(config: JavaExceptionConfig): ExpandedTransform {
+  return (field: ExpandedFormatField): ExpandedFormatField => {
+    const indent = field.indent
+    const lines = field.current.split('\n')
+    const formatted = []
+    for (const line of lines) {
+      if (!/\s/.test(line[0])) {
+        formatted.push(`<span class="has-text-danger">${line}</span>`)
+      } else {
+        const match = line.match(/^(\s+)at (.*)\((.*?)(:\d+)?\)$/)
+        if (!match) {
+          formatted.push(line)
+          continue
+        }
+        const [, space, fqcn, file, lineno] = match
+        let right = `${fqcn}(${file}${lineno || ""})`
+        // TODO: This will have to be customisable when we open source
+
+        const exceptionMatch = fqcnMatches(fqcn, config)
+
+        if (exceptionMatch) {
+          const parts = fqcn.split('.')
+          const firstCapital = parts.findIndex((i: string) => i[0].toLowerCase() !== i[0])
+          const path = parts.slice(0, firstCapital).join('/')
+          const href = exceptionMatch.href
```

Comment:
so `href` is a "template string" that you pass in via a config somewhere? Tests and/or documentation would be really useful here.

Reply:
I have a ticket for updating the README about this. Similar with the other comment: https://github.com/cashapp/logquacious/pull/10/files/022670f93aa7feef2d7a60f2a7cb9a7879eb4b92#r348552561